addons/portal/static/src/js/portal_chatter.js

```diff
@@ -211,6 +211,12 @@ var PortalChatter = publicWidget.Widget.extend({
         if (self.options['token']) {
             data['token'] = self.options['token'];
         }
+        if (self.options['hash'] && self.options['pid']) {
+            Object.assign(data, {
+                'hash': self.options['hash'],
+                'pid': self.options['pid'],
+            });
+        }
```

Comment:
Was there an extra fix in 16.3? For init and message fetch as well?

Reply:
No, I haven't seen any fix for that in 16.3. The only module that uses `hash` and `pid` for authentication in the current portal chatter code is `website_slide` and it needs to fetch avatars too.

Reply:
But this diff is in `_messageFetchPrepareParams`, it's not related to avatar?

Reply:
When we fetch the messages, we need to fetch the avatars too and the new `portal_avatar` controller needs authentication.

Reply:
[getting the avatar url in _portal_message_format ](https://github.com/odoo/odoo/blob/57b05eb1076a9399bc34c396c7a0e3371db1c1e9/addons/portal/models/mail_message.py#L97)

Reply:
Ok I see, computing the avatar url in python rather than in JS actually requires an extra step compared to v16, but I guess it's fine.